From a26cb4c3163fa361675dd024c720625f015a36b9 Mon Sep 17 00:00:00 2001
From: Shreyash Damanwala <shreyash.damanwala@nxp.com>
Date: Fri, 21 Aug 2026 14:58:17 +0530
Subject: [PATCH] arm64: dts: imx95: Add no-display DPU overlay for PCIe iKVM
 endpoint

Add imx95-19x19-evk-no-display.dtso to enable DPU probe without any display bridge for PCIe iKVM headless endpoint mode. The DPU KMS driver skips bridge lookup when pixel_interleaver is disabled, allowing the DPU to probe.

Upstream-Status: Pending
Signed-off-by: Shreyash Damanwala <shreyash.damanwala@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  6 ++++
 .../freescale/imx95-19x19-evk-no-display.dtso | 30 +++++++++++++++++++
 2 files changed, 36 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx95-19x19-evk-no-display.dtso

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index ca33992bc568..e927cf59ec2c 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -1005,6 +1005,9 @@ dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-os08a20-dual.dtb
 imx95-19x19-evk-adv7535-dtbs := imx95-19x19-evk.dtb imx95-19x19-evk-adv7535.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-adv7535.dtb
 
+imx95-19x19-evk-no-display-dtbs := imx95-19x19-evk.dtb imx95-19x19-evk-no-display.dtbo
+dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-no-display.dtb
+
 imx95-19x19-evk-dsi-serdes-dtbs := imx95-19x19-evk.dtb imx95-19x19-evk-dsi-serdes.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-dsi-serdes.dtb
 
@@ -1060,6 +1063,9 @@ dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-ep-adv7535.dtb
 imx95-19x19-evk-ep-mipi-4K-lt9611uxc-dtbs := imx95-19x19-evk.dtb imx95-19x19-evk-lt9611uxc.dtbo imx95-19x19-evk-pcie1-ep.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-ep-mipi-4K-lt9611uxc.dtb
 
+imx95-19x19-evk-ep-no-display-dtbs := imx95-19x19-evk.dtb imx95-19x19-evk-no-display.dtbo imx95-19x19-evk-pcie1-ep.dtbo
+dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-ep-no-display.dtb
+
 imx95-19x19-evk-jdi-wuxga-lvds-panel-dtbs := imx95-19x19-evk.dtb imx95-19x19-evk-jdi-wuxga-lvds-panel.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-jdi-wuxga-lvds-panel.dtb
 
diff --git a/arch/arm64/boot/dts/freescale/imx95-19x19-evk-no-display.dtso b/arch/arm64/boot/dts/freescale/imx95-19x19-evk-no-display.dtso
new file mode 100644
index 000000000000..3f53b54ef315
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx95-19x19-evk-no-display.dtso
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2026 NXP
+ *
+ * imx95-19x19-evk-no-display.dtso
+ *
+ * Device tree overlay for i.MX95 19x19 EVK:
+ *   DPU probe without any display bridge (no-display).
+ *
+ */
+
+/dts-v1/;
+/plugin/;
+
+#include "imx95-clock.h"
+
+&dpu {
+	assigned-clocks = <&scmi_clk IMX95_CLK_DISP1PIX>,
+			  <&scmi_clk IMX95_CLK_VIDEOPLL1_VCO>,
+			  <&scmi_clk IMX95_CLK_VIDEOPLL1>;
+	assigned-clock-parents = <&scmi_clk IMX95_CLK_VIDEOPLL1>;
+	assigned-clock-rates = <0>, <4008000000>, <445333334>;
+	memory-region = <&vram>;
+	status = "okay";
+};
+
+&pixel_interleaver {
+	status = "disabled";
+};
+
-- 
2.34.1

